mobile/src/main/java/com/gelakinetic/mtgfam/fragments/DecklistFragment.java

```diff
@@ -167,188 +233,171 @@ public boolean onItemLongClick(AdapterView<?> parent, View view, int position, l
+                    firstIsi.mNumberOf++;
+                    added = true;
+                    break;
                 }
-            } else {
-                mCompressedDecklist.add(new CompressedDecklistInfo(card, isSideboard));
             }
+            if (!added) {
+                firstCard.add(card);
+            }
+        } else {
+            mCompressedDecklist.add(new DecklistHelpers.CompressedDecklistInfo(card, isSideboard));
+        }
 
-            /* Sort the decklist */
-            Collections.sort(mCompressedDecklist, mDecklistChain);
-
+        /* The headers shouldn't (and can't) be sorted */
+        clearHeaders();
 
-            /* Save the decklist */
-            DecklistHelpers.WriteCompressedDecklist(getActivity(), mCompressedDecklist);
+        /* Load the card's price */
+        if (mShowTotalDecklistPrice) {
+            loadPrice(card.mName, card.setCode, card.mNumber);
+        }
 
-            /* Clean up for the next add */
-            mNumberField.setText("1");
-            mNameField.setText("");
+        /* Sort the decklist */
+        Collections.sort(mCompressedDecklist, mDecklistChain);
 
-            /* Update the headers */
-            setHeaderValues();
+        /* Save the decklist */
+        DecklistHelpers.WriteCompressedDecklist(getActivity(), mCompressedDecklist);
 
-            /* Redraw the new decklist with the new card */
-            mDecklistAdapter.notifyDataSetChanged();
+        /* Clean up for the next add */
+        mNumberOfField.setText("1");
+        mNameField.setText("");
 
-        } catch (FamiliarDbException e) {
-            handleFamiliarDbException(false);
-        } catch (NumberFormatException e) {
-            /* eat it */
+        /* Uncheck the foil box if it isn't locked */
+        if (!mCheckboxFoilLocked) {
+            mCheckboxFoil.setChecked(false);
         }
+
+        /* Redraw the new decklist with the new card */
+        setHeaderValues();
+        mListAdapter.notifyDataSetChanged();
+
     }
 
     /**
-     * read and compress the wishlist
+     * Total the cards, and set it. Get the setting for price. And load up the decklist.
      */
     @Override
     public void onResume() {
+
         super.onResume();
+        final String totalCards = ((CardDataAdapter) mListAdapter).getTotalCards() + " ";
+        mDeckCards.setText(totalCards);
+        mPriceSetting = Integer.parseInt(getFamiliarActivity().mPreferenceAdapter.getTradePrice());
+        mShowTotalDecklistPrice = getFamiliarActivity().mPreferenceAdapter
+                .getShowtotalDecklistPrice();
         mCompressedDecklist.clear();
         readAndCompressDecklist(null, mCurrentDeck);
-        mDecklistAdapter.notifyDataSetChanged();
-    }
+        mListAdapter.notifyDataSetChanged();
 
-    @Override
-    public void onPause() {
-        super.onPause();
-        DecklistHelpers.WriteCompressedDecklist(getActivity(), mCompressedDecklist);
```

Comment:
This is true. There is some data processing in WriteCompressedDecklist() on each save. It's fine now, but if we ever add more complex logic there, saving on each operation could become noticeable. I'm fine with it for now.